doc/news/changes/incompatibilities/20201121Fehling

```diff
@@ -0,0 +1,24 @@
+Deprecated: The template arguments of the following classes have been
+changed to avoid the legacy argument `DoFHandlerType`:
+<ul>
+  <li> `SolutionTransfer<dim, VectorType, DoFHandlerType> -> SolutionTransfer<dim, VectorType, spacedim>`
+  <li> `parallel::distributed::SolutionTransfer<dim, VectorType, DoFHandlerType> -> parallel::distributed::SolutionTransfer<dim, VectorType, spacedim>`
+  <li> `Functions::FEFieldFunction<dim, DoFHandlerType, VectorType> -> Functions::FEFieldFunction<dim, VectorType, spacedim>`
+  <li> `DataOut<dim, DoFHandlerType> -> DataOut<dim, spacedim>`
+  <li> `DataOut_DoFData<DoFHandlerType, patch_dim, patch_space_dim> -> DataOut_DoFData<dim, patch_dim, spacedim, patch_spacedim>`
```

Comment:
I would merge the two change-log entries.

Reply:
This changelog entry is meant for `9.x.y -> 10.0.0` in which we actually want to get rid of all `DoFHandlerType` templates.

The other changelog entry is meant for an intermediate release `9.2.0 -> 9.x.y` to announce that the template argument will be changed soon, so that users can prepare their code to run with the `Legacy` classes if they are not willing to change the template arguments all at once.

If we do both introduction of `Legacy` namespace and deprecation in one release, I'll agree with you to merge both changelogs. It depends on what the principal developers think about how to deal with this. We could discuss this in #11222.

Anyways, my intention was to keep this PR up as WIP until the `10.0.0` release.